### PR TITLE
Add UseNServiceBus to new .NET 8 IHostApplicationBuilder interface

### LINKED ISSUE #429: [.NET 8] Add Extension to UseNServiceBus with the new generic IHostApplicationBuilder
### Describe the feature.

.NET 8 appears to be shipping with an extension to the new generic "minimal api" approach. This will enable a more consistent interface for hosting both APIs and Worker services (see https://github.com/dotnet/runtime/issues/85486). It would be great to introduce an extension on IHostApplicationBuilder to permit initialising and starting up a NServiceBus host, as an alternative approach to the previous [Generic Host](https://learn.microsoft.com/en-us/dotnet/core/extensions/generic-host) method. Typical ceavets would apply, such as not being able to initialise or run the `IHostApplicationBuilder.UseNServiceBus` approach more than once.

Example usage would look like

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by kylejuliandev**
Fixes #429 

Add new extension method for adding NServiceBus to the new .NET 8 Generic host builder approach

Works for WebApplication.CreateBuilder and Host.CreateApplicationBuilder.

### CODE REVIEW

**Comment by bording** [comment]
Hi @kylejuliandev,

Thank you for your contribution. We have routed the PR to the appropriate team and they will get back to you as soon as possible.

Kind regards,
The team in Particular

File: src/NServiceBus.Extensions.Hosting/HostApplicationBuilderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    /// </summary>
+    public static class HostApplicationBuilderExtensions
+    {
+        /// <summary>
+        /// Configures the host to start an NServiceBus endpoint.
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="configureEndpoint"></param>
+        /// <returns></returns>
+        public static IHostApplicationBuilder UseNServiceBus(this IHostApplicationBuilder builder, Func<IHostApplicationBuilder, EndpointConfiguration> configureEndpoint)
```
Comment (kylejuliandev) [inline src/NServiceBus.Extensions.Hosting/HostApplicationBuilderExtensions.cs:21]: Unsure on `Func<IHostApplicationBuilder, EndpointConfiguration>`, possibly `Func<EndpointConfiguration>`, would be more appropriate given you can already access the application builder.

Comment (bording) [reply src/NServiceBus.Extensions.Hosting/HostApplicationBuilderExtensions.cs:21]: I agree that `Func<EndpointConfiguration>` is the better signature here given that there's no special context class to pass in.

**Comment by bording** [comment]
@kylejuliandev Are you still interested in working on this PR? I do think this is a change we want to take, so if you are still interested, go ahead and push up that `Func` change, and we should be good to merge it.

File: src/NServiceBus.Extensions.Hosting/HostApplicationBuilderExtensions.cs (hunk 1 above)
Comment (kylejuliandev) [reply src/NServiceBus.Extensions.Hosting/HostApplicationBuilderExtensions.cs]: I've amended the changes to use the Func<out> approach, hopefully it makes for a clearer abstraction 😄

**Comment by kylejuliandev** [comment]
Hi @bording, have sync'd my fork and amended the changes to amend the signature

**Review by bording — APPROVED** [review]

**Comment by bording** [comment]
@kylejuliandev Thanks for the contribution!

**Pull request merged**